kyaml/kyaml_test.go

```diff
@@ -0,0 +1,1956 @@
+	UintMapSlice    map[string][]uint      `json:"uintMapSlice,omitempty,omitzero"`
+	Uint8MapSlice   map[string][]uint8     `json:"uint8MapSlice,omitempty,omitzero"`
+	Uint16MapSlice  map[string][]uint16    `json:"uint16MapSlice,omitempty,omitzero"`
+	Uint32MapSlice  map[string][]uint32    `json:"uint32MapSlice,omitempty,omitzero"`
+	Uint64MapSlice  map[string][]uint64    `json:"uint64MapSlice,omitempty,omitzero"`
+	Float32MapSlice map[string][]float32   `json:"float32MapSlice,omitempty,omitzero"`
+	Float64MapSlice map[string][]float64   `json:"float64MapSlice,omitempty,omitzero"`
+	TimeMapSlice    map[string][]time.Time `json:"timeMapSlice,omitempty,omitzero"`
+
+	// Map of string to maps
+	StringMapMap  map[string]map[string]string    `json:"stringMapMap,omitempty,omitzero"`
+	BoolMapMap    map[string]map[string]bool      `json:"boolMapMap,omitempty,omitzero"`
+	IntMapMap     map[string]map[string]int       `json:"intMapMap,omitempty,omitzero"`
+	Int8MapMap    map[string]map[string]int8      `json:"int8MapMap,omitempty,omitzero"`
+	Int16MapMap   map[string]map[string]int16     `json:"int16MapMap,omitempty,omitzero"`
+	Int32MapMap   map[string]map[string]int32     `json:"int32MapMap,omitempty,omitzero"`
+	Int64MapMap   map[string]map[string]int64     `json:"int64MapMap,omitempty,omitzero"`
+	UintMapMap    map[string]map[string]uint      `json:"uintMapMap,omitempty,omitzero"`
+	Uint8MapMap   map[string]map[string]uint8     `json:"uint8MapMap,omitempty,omitzero"`
+	Uint16MapMap  map[string]map[string]uint16    `json:"uint16MapMap,omitempty,omitzero"`
+	Uint32MapMap  map[string]map[string]uint32    `json:"uint32MapMap,omitempty,omitzero"`
+	Uint64MapMap  map[string]map[string]uint64    `json:"uint64MapMap,omitempty,omitzero"`
+	Float32MapMap map[string]map[string]float32   `json:"float32MapMap,omitempty,omitzero"`
+	Float64MapMap map[string]map[string]float64   `json:"float64MapMap,omitempty,omitzero"`
+	TimeMapMap    map[string]map[string]time.Time `json:"timeMapMap,omitempty,omitzero"`
+
+	// Recursive types
+	Self           *OmitEmptyAndZeroStruct                       `json:"self,omitempty,omitzero"`
+	SelfSlice      []*OmitEmptyAndZeroStruct                     `json:"selfSlice,omitempty,omitzero"`
+	SelfMap        map[string]*OmitEmptyAndZeroStruct            `json:"selfMap,omitempty,omitzero"`
+	SelfSliceSlice [][](*OmitEmptyAndZeroStruct)                 `json:"selfSliceSlice,omitempty,omitzero"`
+	SelfSliceMap   []map[string]*OmitEmptyAndZeroStruct          `json:"selfSliceMap,omitempty,omitzero"`
+	SelfMapSlice   map[string][]*OmitEmptyAndZeroStruct          `json:"selfMapSlice,omitempty,omitzero"`
+	SelfMapMap     map[string]map[string]*OmitEmptyAndZeroStruct `json:"selfMapMap,omitempty,omitzero"`
+}
+
+func TestKYAMLEncoderFuzzRoundTrip(t *testing.T) {
+	for i := 0; i < 1000; i++ {
+		// Create and fill an instance.
+		original := &AllTypesStruct{}
+		f := randfill.New().NilChance(0.5).NumElements(1, 3).MaxDepth(3)
```

Comment:
Does this actually need to be allocated in the hot loop? Looks to me like we could hoist `randfill.New()` (though I'm not sure if it's concurrent-access safe if we parallelize the iterations, the usage docs don't mention)